src/components/ValueInput/ValueInput.js

```diff
@@ -156,24 +156,29 @@ export const ValueInputComponent = (props: Props) => {
   const formattedMaxValueInFiat = getFormattedBalanceInFiat(fiatCurrency, maxValue, ratesWithCustomRates, assetSymbol);
   const formattedValueInFiat = getFormattedBalanceInFiat(fiatCurrency, value, ratesWithCustomRates, assetSymbol);
 
+  React.useEffect(() => {
+    if (disabled) { // handle fiat updates when disabled, e.g. on Exchange screen
+      const fiatValue = getBalanceInFiat(fiatCurrency, value, ratesWithCustomRates, assetSymbol);
+      setValueInFiat(String(fiatValue ? fiatValue.toFixed(2) : 0));
+    }
+  }, [value]);
+
   const handleValueChange = (newValue: string) => {
-    let errorMessage = null;
     // ethers will crash with commas, TODO: we need a proper localisation
     newValue = newValue.replace(/,/g, '.');
     if (displayFiatAmount) {
-      setValueInFiat(newValue);
-      const convertedValue =
+      const split = newValue.split('.');
+      // only allow 2 decimals in fiat mode
+      if (split.length <= 2 && split[1] ? split[1].length <= 2 : true) {
```

Comment:
maybe it'd be bettter to avoid nested if like this?
`if(split.length <= 2 && !(split[1] && split[1].length > 2) {`